RoutesApp_ios/modules/home/controllers/HomeViewController.swift

```diff
@@ -98,6 +99,56 @@ class HomeViewController: UIViewController {
         self.mapView.animate(toZoom: zoom)
     }
 
+    @IBAction func backButtonAction(_ sender: Any) {
+        switch viewmodel.pointsSelectionStatus {
+        case.pendingOrigin:
+            return
+        case.pendingDestination:
+            labelHelper.text = String.localizeString(localizedString: "select-origin")
+            viewmodel.pointsSelectionStatus = .pendingOrigin
+            viewmodel.origin?.map = mapView
+            viewmodel.origin?.map = nil
+            viewmodel.origin = nil
+            backButton.isHidden = true
+        case.bothSelected:
+            labelHelper.text = String.localizeString(localizedString: "select-destination")
+            viewmodel.pointsSelectionStatus = .pendingDestination
+            viewmodel.destination?.map = mapView
+            viewmodel.destination?.map = nil
+            viewmodel.destination = nil
+        }
+
+    }
+
+    @IBAction func continueButtonAction(_ sender: Any) {
+        let position = mapView.camera.target
+        let pos = GMSMarker(position: position)
+
+        switch viewmodel.pointsSelectionStatus {
+        case.pendingOrigin:
+            pos.title = String.localizeString(localizedString: "origin")
+            labelHelper.text = String.localizeString(localizedString: "select-destination")
+            pos.icon = UIImage(named: "origin_point")
+            pos.map = mapView
+            viewmodel.origin = pos
+            viewmodel.pointsSelectionStatus = .pendingDestination
+
+        case.pendingDestination:
+            pos.title = String.localizeString(localizedString: "destination")
```

Comment:
same in here add destination as constant string

Reply:
changed